### PR TITLE
libuuid-util-linux: new port

### PR DESCRIPTION
**Pull request by dgilman**
#### Description
Build the libuuid that's shipped with util-linux. The name of this package is kind of a mouthful, if there is a more straightforward way to describe it, I'm all ears.

It conflicts with all the other uuid libraries available in MacPorts. I have started some of the work to move them over, but because all of these libraries are going to conflict with each other when that happens, it's best that they all get flipped in one big PR.  So, for now, that can wait.

###### Type(s)
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 10.14.6 18G6020
Xcode 11.3.1 11C504

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by kurthindenburg** [comment]
1. Wouldn't util-linux-libuuid be better?
2. What's the reason for making this a sub-port instead of just enabling it in the main port?

**Comment by ryandesign** [comment]
What's the benefit to having util-linux's libuuid, vs the existing libuuid and ossp-uuid ports we have?

For the longest time, ossp-uuid was the only uuid we had in MacPorts, but Linux had libuuid and a lot of software wouldn't work with ossp-uuid. Now that we have libuuid, I had hoped that we would move toward making libuuid the default uuid library in MacPorts, move ossp-uuid so that it installs into a different place, and move all the ports that can use it over to libuuid. Adding util-linux's libuuid into the mix is a complication and I would rather only do so if there is a clear reason to (e.g. if there is some software out there that works with util-linux's libuuid that doesn't work with the existing libuuid or ossp-uuid).

**Comment by dgilman** [comment]
I can change the name to whatever, you can have the final say on that Kurt.

Because the libuuid from this package has to conflict with the other two libuuid packages it makes sense to me to keep it out of the main port. Because none of the other ports have yet been moved onto this libuuid it would make it impossible for anyone to install util-linux and any of those other uuid-depending ports at the same time, probably breaking a lot of people.

The MacPorts libuuid is a fork of util-linux libuuid. It is also unmaintained upstream. UUID libraries aren't a hotbed of development activity but the util-linux libuuid is getting fixes, documentation updates and even occasional optimizations.

I think there is a strong argument for dropping the existing libuuid from macports entirely and shipping this package's libuuid as the macports libuuid. It may just not be the first step that we want to take for the introduction of this package.

File: devel/util-linux/Portfile [hunk 1]
```diff
@@ -100,4 +102,21 @@ destroot {
...
+    license                 BSD
+    description             Universally unique id library
+    long_description        ${description} from util-linux.
+    conflicts               ossp-uuid libuuid
+    configure.args          --disable-all-programs \
+                            --enable-libuuid \
+                            --disable-bash-completion
+    build.target            libuuid
+    destroot {
+        system -W ${worksrcpath} "make install DESTDIR=${destroot}"
```
Comment (ryandesign) [inline devel/util-linux/Portfile:117]: Why is this needed? Isn't this what the destroot phase would do normally if you didn't override it? So why override it? Why not put the `file delete --force`s below (which could be simplified to just `delete`s) in a `post-destroot` block? Put all of the code above this subport which pertains only to the main port inside an `if {${subport} ne ${namme}} { ... }` block.

**Review by ryandesign — COMMENTED** [review]
> The MacPorts libuuid is a fork of util-linux libuuid.

I agree, their first commit message says it's an import from util-linux: https://sourceforge.net/p/libuuid/code/ci/0b001d78d7e66f90abca32c6258b627f68ad0538/

> It is also unmaintained upstream.

I agree, their last commit was in 2014.

> I think there is a strong argument for dropping the existing libuuid from macports entirely and shipping this package's libuuid as the macports libuuid.

I agree, I'd suggest doing that. If the major library version has changed, or if the API/ABI is different in some way, everything depending on libuuid would need a revbump.
